### PR TITLE
Map return values of inlined AOT functions.

### PR DESCRIPTION
**Pull request by ptersilie**

### CODE REVIEW

File: tests/c/simple_inline.newcg.c [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+//     ...
+//     --- End jit-pre-opt ---
+//     foo 6
+//     jitstate: enter-jit-code
+//     foo 5
+//     foo 4
+//     jitstate: deoptimise
+//     exit
+
+// Check that basic trace compilation works.
```
Comment (vext01) [inline tests/c/simple_inline.newcg.c:24]: This comment and the one below are inaccurate, right?

File: ykrt/src/compile/jitc_yk/trace_builder.rs [hunk 2]
```diff
@@ -487,10 +491,18 @@ impl<'a> TraceBuilder<'a> {
...
     ) -> Result<(), CompilationError> {
         // FIXME: Map return value to AOT call instruction.
-        self.frames.pop();
+        let frame = self.frames.pop().unwrap();
+        if let Some(val) = val {
+            match self.handle_operand(val)? {
+                jit_ir::Operand::Local(jitidx) => {
+                    self.local_map.insert(frame.callinst.unwrap(), jitidx);
+                }
+                _ => todo!(),
```
Comment (vext01) [inline ykrt/src/compile/jitc_yk/trace_builder.rs:503]: Next PR: return a constant?

**Comment by vext01** [comment]
Just a couple of comments.

File: ykrt/src/compile/jitc_yk/trace_builder.rs (hunk 2 above)
Comment (ptersilie) [reply ykrt/src/compile/jitc_yk/trace_builder.rs:503]: Yes, that's the plan.

File: tests/c/simple_inline.newcg.c (hunk 1 above)
Comment (ptersilie) [reply tests/c/simple_inline.newcg.c]: [9115a7e](https://github.com/ykjit/yk/pull/1142/commits/9115a7e72d19eb383841ef20e55cfd367a369429)

**Comment by vext01** [comment]
Please squash

**Comment by ptersilie** [comment]
Squashed.

**Pull request merged**